### PR TITLE
Add Share Modal with QR Codes

### LINKED ISSUE #4: Feature Request: QR-Codes for Rooms
Wenn opening a room, it would be very convenient to display an automatically generated QR-Code which contains the Room-URL. 

With so many readily available QR-Code-Libraries I guess this should be fairly trivial and a quick win, but I sadly lack the experience to implement this reasonably fast.

If no one wants to implement this, I would have a go at it later some time.

### PR DESCRIPTION
**Pull request by Laurendus**
closes #4

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -127,6 +128,27 @@ if (!oauthEnabled) {
...
+  let path: string;
+  if (req.params.type === "join") {
+    path = 'stimmung';
+  } else {
+    res.sendStatus(400);
+    return;
+  }
+
+  res.set('Content-Type', 'image/png');
+  res.send(await QRCode.toBuffer(`${process.env.OAUTH_REDIRECT_BASE || req.hostname}/${path}/${req.params.room || ''}`, options));
```
Comment (acran) [inline index.ts:149]: `req.hostname` does not include protocol (and port) and thus results in an invalid URL if `OAUTH_REDIRECT_BASE` is not set.

File: index.ts [hunk 2]
```diff
@@ -127,6 +128,27 @@ if (!oauthEnabled) {
   });
 }
 
+app.get("/qr/:type/:room?", async (req: express.Request, res: express.Response) => {
+  const options: QRCode.QRCodeToBufferOptions = {
+    width: 300,
+    color: {
+      dark: "000000ff",
+      light: "ffffff00"
```
Comment (acran) [inline index.ts:136]: The background color is white, but transparent. A solid white background better allows the image to be copied:
```suggestion
      light: "ffffff"
```

**Review by acran — CHANGES_REQUESTED** [review]

File: index.ts (hunk 1 above)
Comment (Laurendus) [reply index.ts:149]: Oh. I guess I read the documentation wrong. My reasoning was that if OAuth is not being used, we would need a way to determine the URL. I actually forgot about the protocol (`req.secure ? 'https' : 'http'`) but thought the port would be included in `req.hostname`.

File: index.ts (hunk 2 above)
Comment (Laurendus) [reply index.ts:136]: According to the documentation it needs to be an rgba-value, but my initial reasoning for the became invalid by introducing the sharing modal.

File: README.md [hunk 3]
```diff
@@ -20,6 +20,10 @@ Run ```npm start```.
 * ```HTML_AUTHOR```: The HTML author of the app.
 * ```HTML_DESCRIPTION```: The HTML description of the app.
 * ```NAME_PATTERN```: The pattern the names have to match to join a session. Default is ```.*```
+* ```BASE_URL```: The base URL containing request protocol (http | https), domain and optionally port, eg. ```https://example.com:8443```
```
Comment (acran) [inline README.md:23]: This makes `OAUTH_REDIRECT_BASE` obsolete and it should be removed below in [line 34](https://github.com/BDSU/meeting-moderation-cards/pull/9/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R34)

**Pull request merged**